Added following Files:

roshan-spryiq opened the pull request:
1) connector.py
2) constants.py
3) info.json
4) microsoft_api_auth.py
5) operations.py
AND ADDED SAMPLE playbooks.json AND IMAGES

### Descriptions:
_Add PR descriptions here_

#### Fix:
_Add fix/changes here_

#### Affected Actions:
- [x] All
- [ ] Check Health
- [ ] Other Actions

#### QA Impact:
- [x] `full`
- [ ] Test what is prescribed in the Mantis Ticket
- [ ] Regression Impacts
- [ ] Regression Area X
- [ ] Ensure that platform dependencies remain unaffected

#### UTCs:
- [ ] Connector installation verified
- [ ] Connector logo verified
- [ ] Check health verified
- [ ] Docs link verified
- [ ] Actions and Playbooks list verified
- [ ] Playbook tags verified
- [ ] All playbooks are in info mode verified
- [ ] All playbooks are in inactive mode verified
- [ ] Ingestion playbooks are verified
- [ ] Verified platform dependencies remain unaffected

#### Pytest Test Cases: (If applicable)
- [ ] Attach the test report

roshan-spryiq committed:
Added following Files:
1) connector.py
2) constants.py
3) info.json
4) microsoft_api_auth.py
5) operations.py
AND ADDED SAMPLE playbooks.json AND IMAGES

roshan-spryiq committed:
Create README.md

roshan-spryiq committed:
Update README.md

roshan-spryiq committed:
Updated README.md

roshan-spryiq committed:
Update README.md

Added detailed information on "permissions required"

roshan-spryiq committed:
Updated README.md

roshan-spryiq committed:
Added following Files:
1) connector.py
2) constants.py
3) info.json
4) microsoft_api_auth.py
5) operations.py
AND ADDED SAMPLE playbooks.json AND IMAGES

roshan-spryiq committed:
Merge remote-tracking branch 'origin/develop' into develop

roshan-spryiq committed:
Added following Files:
1) connector.py
2) constants.py
3) info.json
4) microsoft_api_auth.py
5) operations.py
AND ADDED SAMPLE playbooks.json AND IMAGES

roshan-spryiq commented:
1) Rearranged the folder structure and moved all the connector fiiles in the folder named: "microsoft-defender-threat-intelligence"
2) Added missing descriptions for the parameters.
3) Added output schema for the operations.

roshan-spryiq committed:
Added following Files:
ADDED README.md in docs folder

roshan-spryiq committed:
Added following Files:
ADDED README.md in docs folder

cs-utkarsh-surana commented on .fortisoar/local_data.json:
```diff
@@ -0,0 +1 @@
+{}
```
Can you remove .fortisoarand LICENSE file

roshan-spryiq replied:
Removed following Folder/Files: '.fortisoar' folder and 'LICENSE' file

cs-utkarsh-surana commented on LICENSE:
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2024 Fortinet Inc
+Copyright (c) 2025 Fortinet Inc
```
Can you remove .fortisoarand LICENSE file

roshan-spryiq replied:
Removed following Folder/Files: '.fortisoar' folder and 'LICENSE' file

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/operations.py:
```diff
@@ -0,0 +1,360 @@
+                'Prefer': 'return=representation'
+            })
+            self.session.verify = self.verify_ssl
+            logger.error("Session set up successfully")
```
change the logger level to debug.

roshan-spryiq replied:
Changed the logger level to debug where ever I am not raising the error.

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/operations.py:
```diff
@@ -0,0 +1,360 @@
+
+def get_host_reputation(config, params):
+    try:
+        logger.error("Starting 'get_host_reputation' operation with inputs: %s", params)
```
change the logger level to debug.

roshan-spryiq replied:
Changed the logger level to debug where ever I am not raising the error.

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/operations.py:
```diff
@@ -0,0 +1,360 @@
+
+def get_host_details(config, params):
+    try:
+        logger.error("Starting 'get_host_reputation' operation with inputs: %s", params)
```
change the logger level to debug where ever you are not raising the error.

roshan-spryiq replied:
Changed the logger level to debug where ever I am not raising the error.

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/info.json:
```diff
@@ -0,0 +1,802 @@
+    "label": "Microsoft Defender Threat Intelligence",
+    "description": "The Microsoft Defender Threat Intelligence FortiSOAR Connector enables automated threat intelligence gathering, enrichment, and response within FortiSOAR",
+    "publisher": "SpryIQ.co",
+    "icon_small_name": "small.png",
```
Include the key "contributor": "Roshan Bangera"

roshan-spryiq replied:
Included the key "contributor": "Roshan Bangera".

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/info.json:
```diff
@@ -0,0 +1,802 @@
+            "category": "containment",
+            "parameters": [
+                {
+                    "title": "Host ID",
```
The title and the description doesn't match please check.

Or you can change the description to 

Unique identifier for the host, which can be an IP address or Fully Qualified Domain Name (FQDN), used to retrieve its security reputation and threat intelligence data.

roshan-spryiq replied:
Changed the description of 'HostId' parameter.

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/info.json:
```diff
@@ -0,0 +1,802 @@
+        },
+        {
+            "title": "Get Host Details",
+            "description": "Unique identifier (IP address or FQDN) used to retrieve detailed host information.",
```
Action description not proper.

roshan-spryiq replied:
Changed the description of 'Get Host Details' action.

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/info.json:
```diff
@@ -0,0 +1,802 @@
+            "category": "containment",
+            "parameters": [
+                {
+                    "title": "Select Host ID or whoisRecord ID",
```
Change the title to --> Get Record By

roshan-spryiq replied:
I have kept the name of the action as mentioned in the API Documentation. For your ready reference: https://learn.microsoft.com/en-us/graph/api/security-whoisrecord-get?view=graph-rest-1.0&tabs=http

cs-utkarsh-surana commented on microsoft-defender-threat-intelligence/info.json:
```diff
@@ -0,0 +1,802 @@
+    "operations": [
+        {
+            "title": "Get Host Reputation",
+            "description": "Retrieves security reputation and threat intelligence data for a specified host to assess its risk level and potential malicious activity.",
```
please make the corresponding changes in the docs file as well.

roshan-spryiq replied:
Updated the corresponding changes in the docs file.

roshan-spryiq committed:
Below are the commit changes done-
1. Removed following Folder/Files: '.fortisoar' folder and 'LICENSE' file.
2. Changed the logger level to debug where ever I am not raising the error.
3. Included the key "contributor": "Roshan Bangera".
4. Changed the description of 'HostId' parameter.
5. Changed the description of 'Get Host Details' action.
6. Updated the corresponding changes in the docs file.
